src/catalog/src/lib.rs

```diff
@@ -198,6 +198,9 @@ pub trait ReadOnlyDurableCatalogState {
     /// it will fail to open in read only mode.
     async fn open_read_only(&mut self) -> Result<(), Error>;*/
 
+    /// Returns true if the catalog is opened in read only mode, false otherwise.
+    fn is_read_only(&self) -> bool;
```

Comment:
Why does this need to exist on the ReadOnly trait? Could it just be `fn is_read_only(&self) -> bool { true }`?

Reply:
Good point, I moved this to the `DurableCatalogState` trait.